event-schema/counter_manager.go

```diff
@@ -14,14 +14,32 @@ func init() {
 	}
 }
 
-func populateFrequencyCounters(schemaHash string, frequencyCounters []*FrequencyCounter) {
+// populateFrequencyCountersBounded is responsible for capturing the frequency counters which
+// are available in the db and store them in memory but in a bounded manner.
+func populateFrequencyCountersBounded(schemaHash string, frequencyCounters []*FrequencyCounter, bound int) {
 	frequencyCountersMap := make(map[string]*FrequencyCounter)
+	count := 0
+
 	for _, fc := range frequencyCounters {
+		// If count exceeds for a particular schema hash, break
+		// the loop
+		if count >= bound {
+			break
+		}
+
 		frequencyCountersMap[fc.Name] = NewPeristedFrequencyCounter(fc)
+		count++
 	}
 	countersCache[schemaHash] = frequencyCountersMap
 }
 
+// populateFrequencyCounters is responsible for capturing the frequency counters
+// for a particular schema version into memory for fast computation.
+func populateFrequencyCounters(schemaHash string, frequencyCounters []*FrequencyCounter) {
+	populateFrequencyCountersBounded(schemaHash, frequencyCounters, frequencyCounterLimit)
+	return
```

Comment:
I am wondering why we have introduced a new method, instead of keeping the name the same or renaming it. 

Since the function is not exported we are not breaking any API

Reply:
Still exploring the event schema domain space. I agree to the point that given they are unexported we can have only one. I'll make the change.